src/Clock.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+type Props = {
+  name: string;
+};
+
+type State = {
+  time: string;
+};
+
+function getData(): string {
+  return new Date().toUTCString().slice(-12, -4);
+}
+
+export class Clock extends React.Component<Props, State> {
+  timerId = 0;
+
+  state: Readonly<State> = {
+    time: getData(),
+  };
+
+  handleTimerChange = () => {
+    this.timerId = window.setInterval(() => {
+      const newTime = getData();
+
+      this.setState({ time: newTime });
+    }, 1000);
+  };
+
+  componentDidMount(): void {
+    this.handleTimerChange();
+  }
+
+  componentDidUpdate(
+    prevProps: Readonly<Props>,
+    prevState: Readonly<State>,
+  ): void {
+    const clockNameChanges = prevProps.name !== this.props.name;
+    const isTimeChanged = prevState.time !== this.state.time;
```

Comment:
You can destructure `prevaProps` `prevState` `this.props` and `this.state` vales 

```suggestion
    const { name } = prevProps;
    const { time } = prevState;
    const { name: propsNmae } = this.props;
    const { time: propsTime } = this.state;
    
    const clockNameChanges = name !== propsName;
    const isTimeChanged = time !== stateTtime;
```